charts/redpanda/templates/services.nodeport.yaml

```diff
@@ -38,42 +38,38 @@ spec:
   ports:
 {{- range $name, $listener := $values.listeners.admin.external }}
   {{- $enabled := dig "enabled" $values.external.enabled $listener }}
-  {{- $type := dig "type" $values.external.type $listener }}
-  {{- if and $enabled (eq $type "NodePort") }}
+  {{- if $enabled }}
     - name: admin-{{ $name }}
       protocol: TCP
       port: {{ $values.listeners.admin.port }}
-      nodePort: {{ $listener.nodePort }}
+      nodePort: {{ index $listener.advertisedPorts 0 }}
```

Comment:
```suggestion
      nodePort: {{ first $listener.advertisedPorts }}
```

Reply:
This is a breaking change. This will need to be templated to ensure the existing users will not have their nodePort redefined.

Reply:
So rather than relying on semver to notify the user of a breaking change, you suggest that we have a template to determine if the user is trying to use a `values.yaml` file from a previous version? I think I may need to reach out and clarify what the template would try to detect here.